agent/provider/src/market/negotiator/builtin/manifest.rs

```diff
@@ -25,28 +25,27 @@ impl NegotiatorComponent for ManifestSignature {
             return Ok(NegotiationResult::Ready { offer });
         }
 
-        let manifest = match demand.get_property::<String>(DEMAND_MANIFEST_PROPERTY) {
+        let data = match demand.get_property::<String>(DEMAND_MANIFEST_PROPERTY) {
             Err(Error::NoKey(_)) => return Ok(NegotiationResult::Ready { offer }),
             Err(e) => return rejection(format!("invalid manifest type: {:?}", e)),
-            Ok(s) => match decode_manifest(s) {
+            Ok(s) => match decode_data(s) {
```

Comment:
What is reason to have two functions instead of one `decode_manifest`?

Reply:
It is actually a source of a bug I realize I did. While ago I noticed that we extract manifest from demand two times. First to check if features are empty, and sedond time to verify signature. I "fixed" it by extracting it once and reusing it later to verify signature, therefore I started verifying decoded manifest (instead of base64 encoded as I should).

This line reveals another issue - I noticed that some `base64` data decoding is done by attempting to decode `base64` only, but manifest has fallback decoding using `snailquote::unescape`:
```
pub fn decode_manifest<S: AsRef<str>>(input: S) -> Result<AppManifest, Error> {
    match decode_base64(&input) {
        Ok(manifest) => Ok(manifest),
        Err(_) => decode_escaped_json(input),
    }
}

/** .. */

fn decode_escaped_json<S: AsRef<str>>(input: S) -> Result<AppManifest, Error> {
    let decoded = snailquote::unescape(input.as_ref())?;
    Ok(serde_json::de::from_str(&decoded)?)
}
```
I had no idea why manifest base64 encoded data has this special fallback decoding so I started using it on other encoded fields (it will not break anything and makes things more consistent).